lib/ansible/modules/file.py

```diff
@@ -983,6 +983,9 @@ def main():
     elif state == 'absent':
         result = ensure_absent(path)
 
+    if not module._diff:
```

Comment:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.